statistics/handle/lru_cache.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"container/list"
+	"sync"
+
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/statistics"
+)
+
+const (
+	typHit    = "hit"
+	typMiss   = "miss"
+	typUpdate = "update"
+	typDel    = "del"
+	typEvict  = "evict"
+	typCopy   = "copy"
+	typTrack  = "track"
+	typTotal  = "total"
+)
+
+// cacheEntry wraps Key and Value. It's the value of list.Element.
+type cacheItem struct {
+	key         int64
+	value       *statistics.Table
+	tblMemUsage *statistics.TableMemoryUsage
+}
+
+// internalLRUCache is a simple least recently used cache
+type internalLRUCache struct {
+	sync.RWMutex
+	capacity int64
+	// trackingCost records the tracking memory usage of the elements stored in the internalLRUCache
+	// trackingCost should be kept under capacity by evict policy
+	trackingCost int64
+	// totalCost records the total memory usage of the elements stored in the internalLRUCache
+	totalCost int64
+	elements  map[int64]*list.Element
+	// cache maintains elements in list.
+	// Note that if the element's trackingMemUsage is 0, it will be removed from cache in order to keep cache not too long
+	cache *list.List
+}
+
+// newInternalLRUCache returns internalLRUCache
+func newInternalLRUCache(capacity int64) *internalLRUCache {
+	if capacity < 1 {
+		panic("capacity of LRU Cache should be at least 1.")
+	}
+	return &internalLRUCache{
+		capacity: capacity,
+		elements: make(map[int64]*list.Element),
+		cache:    list.New(),
+	}
+}
+
+// Get tries to find the corresponding value according to the given key.
+func (l *internalLRUCache) Get(key int64) (*statistics.Table, bool) {
+	var hit bool
+	var r *statistics.Table
+	l.Lock()
+	r, hit = l.get(key)
+	l.Unlock()
+	if hit {
+		metrics.StatsCacheLRUCounter.WithLabelValues(typHit).Inc()
+	} else {
+		metrics.StatsCacheLRUCounter.WithLabelValues(typMiss).Inc()
+	}
+	return r, hit
+}
+
+func (l *internalLRUCache) get(key int64) (*statistics.Table, bool) {
+	element, exists := l.elements[key]
+	if !exists {
+		return nil, false
+	}
+	l.cache.MoveToFront(element)
```

Comment:
Should we maintain the lru list for Column instead of Table?

Reply:
I think we can't, because the basic recent used element is `table` for current code logic